Matrix.h

```diff
@@ -83,43 +61,54 @@ class Matrix {
 			}
 		}
 
-		nRows = d.size();
-		nCols = checkNumberOfColumns;
+		setFields(d.size(), checkNumberOfColumns);
+		name = UNAMED;
 		data = d;
+	}
 
-		topLimit = 0;
-		bottomLimit = nRows;
-		leftLimit = 0;
-		rightLimit = nCols;
-
+	// Copy constructor copies values between the input Matrix's limits.
+	// @param const Matrix& mat - Matrix from which to copy
+	Matrix(const Matrix& mat) {
+		setFields(mat.bottomLimit - mat.topLimit, mat.rightLimit - mat.leftLimit);
 		name = UNAMED;
+		copyBetweenLimits(mat);
 	}
 
-	~Matrix() {
-		nRows = 0;
-		nCols = 0;
-		data = vector<vector<double>>();
 
-		topLimit = 0;
-		bottomLimit = nRows;
-		leftLimit = 0;
-		rightLimit = nCols;
 
+	// ------------------ //
+	// --- Destructor --- //
+	// ------------------ //
+
+	// Destructor sets all fields to their default states and removes the name
+	~Matrix() {
+		setFields();
 		name = "";
 	}
 
-	// name
-	const string& getName() const {
-		return name;
-	}
-	void setName(const string& n) {
-		name = n;
-	}
 
 
+	// -------------------------- //
+	// --- Blow Out Operators --- //

```

Comment:
Change this comment. Call them assignment operators. Other operator overloads of the = sign are not assignment operator in the strict sense because they can only change the data, not the size.